src/sage/groups/perm_gps/permgroup_named.py

```diff
@@ -3435,3 +3437,91 @@ def codegrees(self):
         ret = [self._m * i for i in reversed(range(self._n-1))]
         ret.append((self._n-1)*self._m - self._n)
         return tuple(sorted(ret, reverse=True))
+
+class SmallPermutationGroup(PermutationGroup_generic):
+    r"""
+    A GAP SmallGroup, returned as a permutation group.
+
+    GAP contains a database of groups, each identified by its GAP id.
+    This is a pair ``[n,k]`` consisting of ``n``, the order of the
+    group, and ``k``, an index determining the group specifically.
+    This class can construct the group as a permutation group from this data.
+
+    INPUT:
+
+    - ``order`` -- the order of the group
+
+    - ``gap_id`` -- the numerical index in the GAP Id of the group
+
+    Generators may be obtained through the :meth:`gens` method.
+    These could change for a particular group in later releases
+    of GAP. In many instances the degree of the constructed group
+    ``SmallPermutationGroup(n,k)`` will be a permutation group on
+    `n` letters, but this will not always be true.
+
+    EXAMPLES::
+
+        sage: G = SmallPermutationGroup(12,4); G
+        Group of order 12 and GAP Id 4 as a permutation group
+        sage: G.gens()
+        ((1,2)(3,5)(4,10)(6,8)(7,12)(9,11),
+        (1,3)(2,5)(4,7)(6,9)(8,11)(10,12),
+        (1,4,8)(2,6,10)(3,7,11)(5,9,12))
+        sage: G.character_table()
+        [ 1  1  1  1  1  1]
+        [ 1 -1 -1  1  1 -1]
+        [ 1 -1  1  1 -1  1]
+        [ 1  1 -1  1 -1 -1]
+        [ 2  0 -2 -1  0  1]
+        [ 2  0  2 -1  0 -1]
+        sage: all(SmallPermutationGroup(n,k).id()==[n,k] for n in [1..64] for k in [1..oeis(1)[n]])
+        True
+        sage: H = SmallPermutationGroup(6,1)
+        sage: H.is_abelian()
+        False
+        sage: [H.centralizer(g) for g in H.conjugacy_classes_representatives()]
+        [Subgroup generated by [(1,2)(3,6)(4,5), (1,3,5)(2,4,6)] of (Group of order 6 and GAP Id 1 as a permutation group),
+        Subgroup generated by [(1,2)(3,6)(4,5)] of (Group of order 6 and GAP Id 1 as a permutation group),
+        Subgroup generated by [(1,3,5)(2,4,6), (1,5,3)(2,6,4)] of (Group of order 6 and GAP Id 1 as a permutation group)]
+    """
+
+    def __init__(self, order, gap_id):
+        """
+        Initialize ``self``.
+
+        TESTS::
+
+            sage: TestSuite(SmallPermutationGroup(60,5)).run()
+        """
+        self._n = order
+        self._gap_id = gap_id
+        self._gap_small_group = libgap.SmallGroup(order,gap_id)
+        gap_permutation_group = self._gap_small_group.IsomorphismPermGroup().Image(self._gap_small_group)
+        PermutationGroup_generic.__init__(self, gap_group=gap_permutation_group)
+
+    def _repr_(self):
+        r"""
+        EXAMPLES::
+
+            sage: G = SmallPermutationGroup(12,4); G
+            Group of order 12 and GAP Id 4 as a permutation group
+        """
+        return "Group of order %s and GAP Id %s as a permutation group"%(self._n, self._gap_id)
```

Comment:
Why "Id" and not "id" ?

Reply:
I revised the docstrings following these comments and pushed. Thanks!

Reply:
It still says `Id` instead of `id` in several places